dev/storage_base.h

```diff
@@ -124,8 +124,27 @@ namespace sqlite_orm {
                 return sqlite3_busy_timeout(con.get(), ms);
             }
 
+            /*
+             * Ownership is NOT transferred, i.e. object of ptr 
+             * must not go out of scope for the lifetime of the database.
+             */
+            template<typename T = void>
+            int busy_handler(std::function<int(T *, int)> f, T *ptr) {
+                struct handler_t final {
+                    std::function<int(T *, int)> fn;
+                    T *ptr;
+                } handler = {std::move(f), ptr};
+                return sqlite3_busy_handler(
+                    this->get_connection().get(),
+                    [](void *ptr, int times) -> int {
+                        auto h = *reinterpret_cast<handler_t *>(ptr);
+                        return h.fn(h.ptr, times);
+                    },
+                    &handler);
```

Comment:
`handler` will be destroyed once this function exits. Then its' address will be captured as user data. Then dereferencing it will lead to UB. It is got to be fixed.

Reply:
We can put the struct inside the protected area.